addon/index.js

```diff
@@ -426,4 +415,21 @@ AddonManager.addInstallListener({
   onDownloadFailed: function(install) {
     app.send('addon-install:download-failed', formatInstallData(install));
   }
+};
+AddonManager.addInstallListener(installListener);
```

Comment:
Might make sense to shuffle some things around - i.e. move all the functions & classes to the bottom. For example, I just now noticed the Metrics stuff got inserted between the `updatePrefs` declaration and the actual initialization of prefs. But, there's a lot going on here between prefs, metrics, toolbar button, and addon/frontend message handling